Contrib/AffiliationServer/revenue.fsx

```diff
@@ -0,0 +1,118 @@
+        match arg.Split([| '=' |]) with
+        | [| k; v |] -> Some (k, v)
+        | _ -> None
+
+    let getArg name =
+        Environment.GetCommandLineArgs()
+        |> Array.map tryParseArgument
+        |> Array.choose id
+        |> Array.tryFind (fun (argName, _) -> argName = name )
+        |> Option.map snd
+
+    let getArgOrDefault name defaultValue =
+        getArg name
+        |> Option.defaultValue defaultValue
+
+    let getArgOrFail name =
+        match getArg name with
+        | Some value -> value
+        | None -> failwith $"Mandatory argument {name} was not provided"
+
+module Signature =
+
+    let verifySignature (pubkey:byte[]) (message:byte[]) signature =
+        let ver = System.Security.Cryptography.ECDsa.Create()
+        let mutable bytesRead = 0
+        ver.ImportSubjectPublicKeyInfo(ReadOnlySpan(pubkey) , &bytesRead)
+        ver.VerifyData(message, signature, HashAlgorithmName.SHA256)
+
+    let verifyNotification affiliate pubkey (notification: CoinJoinNotificationRequest) =
+         let payload = Payload(Header.Create(affiliate), notification.Body)
+         verifySignature pubkey (payload.GetCanonicalSerialization()) notification.Signature
+
+open Signature
+open CommandLine
+
+let args = {|
+    rpcCredentials = RPCCredentialString.Parse (getArgOrFail "--connection")
+    network = Network.GetNetwork (getArgOrDefault "--network" "Main")
+    notificationsPath = getArgOrDefault "--path" "."
+    affiliate = getArgOrDefault "--affiliate" "trezor"
+    pubkey = getArgOrFail "--pubkey" |> Convert.FromHexString
+|}
+
+let rpc = RPCClient(args.rpcCredentials, args.network)
+let notificationFiles = Directory.EnumerateFiles (args.notificationsPath)
+
+type NotificationProcessResult =
+     | ApocryphalNotification of CoinJoinNotificationRequest
+     | NonexistentTransaction of uint256 * CoinJoinNotificationRequest
+
+let processNotification notification = result {
+    do! verifyNotification args.affiliate args.pubkey notification
+        |> Result.requireTrue (ApocryphalNotification notification)
+
+    let notificationBody = notification.Body
+    let txId = uint256.Parse(notificationBody.TransactionId)
+    let! tx =
+        rpc.GetRawTransaction(txId, false)
+        |> Result.requireNotNull (NonexistentTransaction (txId, notification))
+
+    let affInputs =
+        notificationBody.Inputs
+        |> Seq.filter (fun x -> x.IsAffiliated && not x.IsNoFee)
+        |> Seq.map (fun x -> x.Amount, x.Prevout)
+        |> List.ofSeq
+
+    let affInputSum = List.sumBy fst affInputs
+    let affShare = int64 (0.003M * decimal affInputSum)
+
+    return! Ok (txId, affInputs, affInputSum, affShare, notification)
+}
+
+let notificationProcessResult =
+    notificationFiles
+    |> Seq.map File.ReadAllText
+    |> Seq.map (fun s -> JsonConvert.DeserializeObject<CoinJoinNotificationRequest>(s, AffiliationJsonSerializationOptions.Settings))
+    |> Seq.map processNotification
```

Comment:
No because this `map`  doesn't return an `option` but a `Result<>` containing the relevant data in case everything is okay and a `NotificationProcessResult` in case of error (I will rename it to make it more clear)